models/productDao.js

```diff
@@ -0,0 +1,34 @@
+const { appDataSource } = require("./data-source");
+
+const showProducts = async (categoryId, limit, offset) => {
+  try {
+    let query = `SELECT
+          products.id AS products_id,
+          products.name AS products_name,
+          products.price,
+          products.thumbnail,
+          products.category_id,
+          categories.name AS category_name
+        FROM products
+        JOIN categories ON products.category_id = categories.id
+        ORDER BY products.id DESC
+      `;
+    if (categoryId !== "0") {
+      query += ` WHERE products.category_id = ?`;
+    }
+    query += ` LIMIT ? OFFSET ?`;
+    if (categoryId !== "0") {
+      return await appDataSource.query(query, [categoryId, limit, offset]);
+    } else {
+      return await appDataSource.query(query, [limit, offset]);
+    }
+  } catch (err) {
+    const error = new Error("INVALID_DATA_INPUT");
+    error.statusCode = 500;
```

Comment:
status code 500은 개발자가 의도하지 않은 에러가 났을 때 사용하는 코드 입니다. 따라서 500이 나왔다는 것은 내가 예외처리를 안 했다는 의미가 되기 때문에 500번이 아닌 400번으로 수정해 주세요.
```suggestion
    error.statusCode = 400;
```